rmf_site_editor/src/site/change_plugin.rs

```diff
@@ -63,9 +62,7 @@ impl<T: Component + Clone + Debug> Plugin for ChangePlugin<T> {
     fn build(&self, app: &mut App) {
         app.add_event::<Change<T>>().add_systems(
             PreUpdate,
-            update_changed_values::<T>
-                .run_if(AppState::in_displaying_mode())
-                .in_set(SiteUpdateSet::ProcessChanges),
+            update_changed_values::<T>.in_set(SiteUpdateSet::ProcessChanges),
```

Comment:
This is a tricky one, similar change is in the [recall plugin](https://github.com/open-rmf/rmf_site/blob/3496425691f9c43b46b6acb59bd257961e80a6f9/rmf_site_editor/src/site/recall_plugin.rs#L35-L41).
The issue we are trying to tackle is that if users wanted to use this plugin in a third app they would need to include the rmf_site_editor `AppState` or the `run_if` function would panic.
The main drawback here is that I believe this changes the system to _always_ run. Now it could be argued that the net difference in this case (and others through this PR) is that they will additionally run in `AppState::MainMenu`, where there are very few entities, pretty much no work to be done and very little CPU usage, so I'd argue it's not a big issue, but still worth considering.

Reply:
I certainly don't mind migrating away from a dependency on the `AppState`.

If anything maybe we should give this its own state tracker to explicitly toggle `ProcessChanges` on/off, but we can save that for later if we benchmark and find that it's eating up more resources than it should.